authors/apps/articles/serializers.py

```diff
@@ -28,3 +31,63 @@ def validate_description(self, value):
                 'The article should not be more than 255 characters'
             )
         return value
+
+
+class RatingSerializer(serializers.ModelSerializer):
+
+    """
+    Serializer for rating an article model
+    """
+    article = serializers.PrimaryKeyRelatedField(
+        queryset=Article.objects.all())
+    score = serializers.IntegerField(
+        required=True)
+    reader = serializers.PrimaryKeyRelatedField(queryset=User.objects.all())
+
+    @staticmethod
+    def update_data(data, slug, user: User):
+        try:
+            article = Article.objects.get(slug__exact=slug)
+        except Article.DoesNotExist:
+            raise NotFound({
+                'error':  "Article does not exist. "
+                "Check provided Article slug."
+            })
+
+        if article.author == user:
+            raise PermissionDenied({
+                "error":
+                "You cannot rate an article that you created."
+            })
+
+        data.update({"article": article.pk})
+        data.update({"reader": user.pk})
+        return data
+
+    def create(self, data):
+        """
+        Method that saves ratings
+        """
+        reader = data.get("reader")
+        article = data.get("article")
+        score = data.get("score")
+        if score > 5 or score < 1:
```

Comment:
@myrdstom There should be no validation of serializer fields in a `create method` rather if you want to validate the score create a method called `validate_score` on the serializer and do the validation from there. e.g 

```
    def validate_score(self, score):
        """Validate score"""
        if score > 5 or score < 1:
            raise serializers.ValidationError('Score must be an integer between 1 and 5.')
        return score
```

Read more about custom serializer validation [here](https://micropyramid.com/blog/custom-validations-for-serializer-fields-django-rest-framework/)

Reply:
Implemented

Reply:
I see the mistake I was making, adjusted the serializer code accordingly